### PR TITLE
helm: add support for configuring distributor

### PR DESCRIPTION
**Pull request by rissson**
**What this PR does / why we need it**:

Add support for configuring distributor with Helm. Needed to be able to enable IPv6 on that component.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] N/A: Tests updated
- [x] `CHANGELOG.md` updated
  - [x] N/A: If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] N/A: Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: production/helm/loki/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ Entries should include a reference to the pull request that introduced the chang
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
 
+## 5.21.0
+
+- [FEATURE] Add support for configuring distributor
```
Comment (JStickler) [inline production/helm/loki/CHANGELOG.md:18]:
```suggestion
- [FEATURE] Add support for configuring distributor to Helm charts
```

**Review by JStickler — COMMENTED** [review]
[docs team] LGTM

**Comment by rissson** [comment]
Is this missing anything?

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**